File: backend/schemas/session.py
# backend/schemas/session.py
#
# Pydantic models for session-related request and response bodies.
# These are the data contracts between frontend and backend for sessions.

from datetime import datetime
from pydantic import BaseModel
from schemas.document import DocumentInfo


class SessionData(BaseModel):
    """Full session state returned by GET /session/{id} and POST /session."""
    session_id:        str
    created_at:        datetime
    expires_at:        datetime
    documents:         list[DocumentInfo]
    total_chunks:      int


class SessionResponse(BaseModel):
    """Standard success envelope wrapping session data."""
    success: bool = True
    data:    SessionData


class DeleteSessionResponse(BaseModel):
    """Response after a session is manually deleted."""
    success: bool = True
    data:    dict  # {"message": "...", "session_id": "..."}

File: frontend/src/api/session.js
// frontend/src/api/session.js
//
// All session-related API calls.
// Each function returns the `data` field of the response envelope
// (i.e., the actual payload, not the full {success, data} wrapper).

import client from './client'

/**
 * Create a new session on the backend.
 * @returns {Promise<object>} Session data: { session_id, created_at, expires_at, ... }
 */
export async function createSession() {
  const response = await client.post('/session')
  return response.data.data
}

/**
 * Fetch the current state of an existing session.
 * Used on page reload to restore the document list.
 * @param {string} sessionId
 * @returns {Promise<object>} Session data including documents array
 */
export async function getSession(sessionId) {
  const response = await client.get(`/session/${sessionId}`)
  return response.data.data
}

/**
 * Permanently delete a session and all its data.
 * @param {string} sessionId
 * @returns {Promise<object>} { message, session_id }
 */
export async function deleteSession(sessionId) {
  const response = await client.delete(`/session/${sessionId}`)
  return response.data.data
}